docs/core/eventdata/tracks.md

```diff
@@ -1,4 +1,730 @@
+
+The backend needs to flag itself as mutable or const, by specializing
+
+```cpp
+template <typename T>
+struct IsReadOnlyMultiTrajectory;
+// or
+template <typename T>
+struct IsReadOnlyTrackContainer;
+```
+
+This informs the interface layer to permit or prevent mutable access.
+
+Common between both track and track state backends is the component access.
+Here, the component is identified by a compile-time string hash of the
+component name, which the backend responds to by a type-erased mutable or const
+pointer. There is a hard requirement for the backend to return stable pointers
+here, as the interface layer expects this and exposes this in the API.
+
+This can be accomplished like in the transient backend with a `switch`-statements like:
+
+```cpp
+switch (key) {
+  case "previous"_hash:
+    return &m_previous[istate];
+  case "next"_hash:
+    return &m_next[istate];
+  case "predicted"_hash:
+    return &m_index[istate].ipredicted;
+  case "filtered"_hash:
+    return &m_index[istate].ifiltered;
+  case "smoothed"_hash:
+    return &m_index[istate].ismoothed;
+  case "projector"_hash:
+    return &m_projectors[instance.m_index[istate].iprojector];
+  case "measdim"_hash:
+    return &m_index[istate].measdim;
+  case "chi2"_hash:
+    return &m_index[istate].chi2;
+  case "pathLength"_hash:
+    return &m_index[istate].pathLength;
+  case "typeFlags"_hash:
+    return &m_index[istate].typeFlags;
+  default:
+    // handle dynamic columns
+}
+```
+
+The `default` statement deals with [](#track_edm_dynamic_columns).  For support
+of dynamic columns, the backend needs to implement `hasColumn_impl` to return
+if a column exists, mutable backends need to implement `addColumn_impl` which
+adds a column with a type and a string name. `copyDynamicFrom_impl` and
+`ensureDynamicColumn_impl` on mutable backends allows copying dynamic content
+between container backends. `dynamicKeys_impl` rteurns an iterator pair that
+informs the caller of which dynamic keys are registered. This is also used in
+dynamic column copies.
+
+Both backend types return parameter vectors and covariance matrices. This is
+always done as `Eigen` maps, which have reference semantics into the backing
+storage.
+
+#### TrackContainer backend
+
+- Mutable & const
+    - Get size
+    - Reference parameter vector and covariance matrix access
+    - Get if column exists
+    - Get reference surface (returns `const Surface*`)
+    - Get particle hypothesis (returns by-value, so allows on-the-fly creation)
+    - Get dynamic keys
+    - Type-erased component access
+- Mutable
+    - Add and remove a track
+    - Add column
+    - Ensure dynamic columns, copy dynamic columns
+    - Reserve number of entries
+    - Set reference surface
+    - Set particle hypothesis
+    - Clear the container
+
+#### MultiTrajectory (track state) backend
+
+The track state container can [share]
+components](#track_edm_component_sharing). This is implemented for the
```

Comment:
```suggestion
The track state container can [share
components](#track_edm_component_sharing). This is implemented for the
```

Reply:
maybe this needs to be on the same line?